packages/react-router-dom/docs/guides/query-string-parsing.md

```diff
@@ -0,0 +1,61 @@
+# Query String Parsing
+
+In earlier versions of React Router we provided out-of-the-box support for query string parsing. Since there are too many different parsing-algorithms out there and people had problems with an opinionated version, it's been removed from React Router since v4.
+
+But it's easy to integrate your favourite query string parsing lib!
+
+First install [`history`](https://www.npmjs.com/package/qs) and the parsing library of your choice (e.g. [querystring](https://www.npmjs.com/package/querystring) or [qs](https://www.npmjs.com/package/qs))
+
+```sh
+yarn add history, querystring -S
+```
+
+Instead of using `BrowserRouter` which has `browserHistory` already included, you'll need to use the basic `Router` and create your custom `browserHistory` object.  
```

Comment:
I think that instead of saying
> which has `browserHistory` already included

it would be better to say

> Instead of using `<BrowserRouter>`, which creates its own `history` object, ...